runtime/runc_jailer.go

```diff
@@ -640,3 +664,52 @@ func (j runcJailer) Stop(force bool) error {
 	}
 	return j.runcClient.Kill(j.ctx, j.vmID, int(signal), &runc.KillOpts{All: true})
 }
+
+// setupCacheTopology will copy indexed contents from the cacheTopologyPath to
+// the jailer. This is needed for arm architecture as arm does not
+// automatically setup any cache topology
+func (j runcJailer) setupCacheTopology(path string) error {
+	j.logger.WithField("path", path).Debug("Creating cache topology")
+	const mode = os.FileMode(0700)
+
+	// builds the cache topology path from the root directory
+	for _, p := range cacheTopologyPaths {
+		path = filepath.Join(path, p)
+		if err := mkdirAndChown(path, mode, j.Config.UID, j.Config.GID); err != nil {
+			return err
+		}
+	}
+
+	err := filepath.Walk(cacheTopologyPath, func(cachePath string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() {
+			return err
+		}
+
+		folder := filepath.Base(cachePath)
+		if !strings.HasPrefix(folder, cacheFolderPrefix) {
+			return err
+		}
```

Comment:
How about returning nil explicitly here?

Reply:
Cool, will fix this. Forgot that I returned if `err` was not `nil` at the top of the closure.